src/mpyl/cli/commands/projects/lint.py

```diff
@@ -124,11 +127,48 @@ def __detail_wrong_substitutions(
         project.name.lower(): project.name for project in all_projects
     }
     for project in wrong_substitutions_per_project:
-        console.print(f"❌ Project {project.name} has wrong namespace substitutions:")
+        console.print(f"  ❌ Project {project.name} has wrong namespace substitutions:")
         for env, url in project.wrong_substitutions:
             unrecognized_project_name = url.split(".{namespace}")[0].split("/")[-1]
             suggestion = all_project_names.get(unrecognized_project_name.lower())
             console.print(
                 f"  {env} references unrecognized project {unrecognized_project_name}"
                 + (f" (did you mean {suggestion}?)" if suggestion else "")
             )
+
+
+def _lint_whitelisting_rules(
+    console: Console,
+    projects: list[Project],
+    config: dict,
+    target: Target,
+):
+    console.print("")
+    console.print("Checking whitelisting rules: ")
+    defined_whitelists: set[str] = set(
+        map(lambda rule: rule["name"], config["whiteLists"]["addresses"])
+    )
+    wrong_whitelists: list[tuple[Project, set]] = []
+    for project in projects:
+        if project.deployment:
+            if traefik := project.deployment.traefik:
```

Comment:
To reduce the level of nesting you could filter out project that have `deployment.traefik` up front.

Reply:
done in [30ce624](https://github.com/Vandebron/mpyl/pull/269/commits/30ce624ecfca9fb61e5b312ca1e6c8961c451b8c)